torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -179,6 +179,12 @@ void ProcessGroupAgent::join() {
       SendWork(workerIds_[dst], Message({}, {}, MessageType::SHUTDOWN)));
   threadPool_.waitWorkComplete();
   listenerThread_.join();
+  // explicitly clean up python objects in PythonRpcHandler before destructing
```

Comment:
If we expect all `RpcAgent` implementations would use the same `PythonRpcHandler`, shall we add this to `RpcAgent` instead of an implementation of `RpcAgent`.